File: adder.rs
pub fn	adder(a: u32, b: u32) -> u32 {
	let mut	result = a ^ b;
	let mut	carry = (a & b) << 1;

	while carry != 0 {
		let	prv_result = result;
		result = result ^ carry;
		carry = (prv_result & carry) << 1;
	}
	return result;
}

File: C#/ready_set_boole/Adder.cs
namespace ReadySetBoole
{
    public static class Adder
    {
        public static uint Add(uint a, uint b)
        {
            uint result = a ^ b;
            uint carry = (a & b) << 1;

            while (carry != 0)
            {
                uint previousResult = result;
                result = result ^ carry;
                carry = (previousResult & carry) << 1;
            }
            return result;
        }
    }
}